packages/keyring/src/pair/index.ts

```diff
@@ -124,8 +128,9 @@ export default function createPair ({ toSS58, type }: Setup, { publicKey, secret
       }
     },
     derive: (suri: string, meta?: KeyringPair$Meta): KeyringPair => {
-      assert(secretKey, 'Cannot derive on a locked keypair');
-
+      if (!secretKey || isLocked(secretKey)) {
```

Comment:
```suggestion
      if (isLocked(secretKey)) {
```

`isLocked` already checks for `!secretKey`